### PR TITLE
Initial cicd_gitlab blueprint.

### PR DESCRIPTION
**Pull request by frankfarzan**
This includes GCP resources, future PR will add K8S resources.

### CODE REVIEW

**Comment by frankfarzan** [comment]
@janetkuo @morgante please review

File: krm/cicd_gitlab/database.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+kind: SQLDatabase
+metadata:
+  name: gitlabhq-production
+  namespace: gitlab # {"$kpt-set":"namespace"}
+spec:
+  instanceRef:
+    name: gitlab-db
+---
+apiVersion: v1
+kind: Secret
```
Comment (janetkuo) [inline krm/cicd_gitlab/database.yaml:61]: Is it safe to put `Secret` in the blueprint?

Comment (frankfarzan) [reply krm/cicd_gitlab/database.yaml]: Nope, this was a placeholder for now, moved to be an imperative command. This is a known ux issue we need to hash out.

File: krm/cicd_gitlab/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+kpt cfg cicd_gitlab set project-id A_PROJECT
+kpt cfg cicd_gitlab set namespace A_NAMESPACE
+```
+
+### Apply the package
+
+Create the SQLDatabase secret:
+
+```
+kubectl -n A_NAMESPACE create secret generic gitlab-db-password-new --from-literal password=<PASSWORD>
```
Comment (janetkuo) [inline krm/cicd_gitlab/README.md:32]: Secret name should be `gitlab-db-password`

**Comment by frankfarzan** [comment]
done.

**Pull request merged**